src/robot/model/filter.py

```diff
@@ -72,6 +77,8 @@ def start_suite(self, suite):
             suite.tests = self._filter(suite, self._included_by_tags)
         if self.exclude_tags:
             suite.tests = self._filter(suite, self._not_excluded_by_tags)
+        if self.skip_tags:
+            suite.tests = self._filter(suite, self._not_skipped_by_tags)
```

Comment:
Correct my implementation of --skip is exactly the same as --exclude.

With reference to your comment on 16th Oct https://github.com/robotframework/robotframework/issues/2087

"If skipping is supported, I think it should also be supported based on tags e.g. like --skip mytag. Tests having skipped tags shouldn't be executed at all, and a dynamically added skip tag should mark test skipped afterwards. In practice --skip would work like --exclude but tests would be shown in statistics."

My understanding of the robot source is limited and I could not see exactly how to implement
what you were asking for in your comment. So Implemented as much as I could in replicating
--tagstatexclude with --tagstatskip. I was hoping that you (or someone else in the robot
development could/would use what I have written as a basses for what you are asking for. I don't
understand the source enough to do this correctly I feel :(